libraries/core-react/src/components/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,235 @@
+  disableFuture?: boolean
+  disableBeforeDate?: Date
+  className?: string
+  dateFormat?: string
+  placeholder?: string
+  readOnly?: boolean
+  popperPlacement?:
+    | 'auto-start'
+    | 'auto'
+    | 'auto-end'
+    | 'top-start'
+    | 'top'
+    | 'top-end'
+    | 'right-start'
+    | 'right'
+    | 'right-end'
+    | 'bottom-end'
+    | 'bottom'
+    | 'bottom-start'
+    | 'left-end'
+    | 'left'
+    | 'left-start'
+  locale?: string
+}
+
+export interface DatePickerRefProps extends ReactDatePickerProps {
+  setBlur: () => void
+  setFocus: () => void
+  setOpen: (open: boolean, skipSetBlur?: boolean) => void
+  isCalendarOpen: () => boolean
+}
+
+const ReactDatePicker = forwardRef<DatePickerRefProps, DatePickerProps>(
+  function DatePicker(
+    {
+      label,
+      value,
+      onChanged,
+      id,
+      disableFuture,
+      disableBeforeDate,
+      className,
+      dateFormat = 'dd.MM.yyyy',
+      placeholder = 'DD.MM.YYYY',
+      readOnly = false,
+      popperPlacement,
+      locale = 'en-gb',
+    },
+    ref,
+  ) {
+    const [date, setDate] = useState(value)
+    const onDateValueChange = useCallback(
+      (date: Date): void => {
+        setDate(date)
+        onChanged?.(date)
+      },
+      [onChanged],
+    )
+
+    return (
+      <Container className={`date-picker ${className}`}>
+        <DateLabel
+          htmlFor={id}
+          className={`date-label`}
+          onClick={(): void => {
+            ref?.setOpen?.(true)
+          }}
+        >
+          <span>{label}</span>
+          <StyledDatepicker
+            ref={ref}
+            locale={locale}
+            selected={date}
+            onChange={onDateValueChange}
+            dateFormat={dateFormat}
+            placeholderText={placeholder}
+            id={id}
+            filterDate={(date: Date): boolean => {
+              if (disableFuture) {
+                return new Date() > date
+              }
+              if (disableBeforeDate) {
+                return date >= disableBeforeDate
+              }
+              return true
+            }}
+            onChangeRaw={(): void => {
+              ref?.setOpen?.(false)
+            }}
+            autoComplete="false"
+            popperPlacement={popperPlacement}
+            renderCustomHeader={(props) => (
+              <PopupHeader {...props} changeDate={onDateValueChange} />
+            )}
+            shouldCloseOnSelect={true}
+            readOnly={readOnly}
+          />
+          <CalendarIcon
+            name="calendar"
+            className="calendar-icon"
+            data={calendar}
+            size={24}
+          />
+        </DateLabel>
+      </Container>
+    )
+  },
+)
+console.log('tokens ---> ', tokens)
+
+const Container = styled.div.attrs<DatePickerToken>(tokens)`
+  width: 100%;
+  max-width: 148px;
+  display: flex;
+  flex-direction: column;
+  position: relative;
+  box-sizing: border-box;
+
+  .react-datepicker {
```

Comment:
These look like the internal class names? Seems risky for maintenance and future updates in terms of breaking changes 🙈 Did you try the [custom class names](https://reactdatepicker.com/#example-custom-calendar-class-name) for the different portions of the datepicker?